runtime/memory.c

```diff
@@ -238,50 +238,47 @@ CAMLexport CAMLweakdef void caml_modify (volatile value *fp, value val)
 */
 CAMLexport void caml_alloc_dependent_memory (value v, mlsize_t nbytes)
 {
-  /* No-op for now */
+  if (nbytes == 0) return;
+  CAMLassert (Is_block (v));
+  if (Is_young (v)){
+    Caml_state->stat_minor_dependent_bytes += nbytes;
+    add_to_dependent_table (&Caml_state->minor_tables->dependent, v, nbytes);
+    Caml_state->minor_dependent_bsz += nbytes;
+    if (Caml_state->minor_dependent_bsz >
+          Bsize_wsize (Caml_state->minor_heap_wsz)
+          / 100 * caml_custom_minor_ratio){
+      caml_request_minor_gc ();
+    }
+  }else{
+    caml_add_dependent_bytes (Caml_state->shared_heap, nbytes);
+    Caml_state->allocated_dependent_bytes += nbytes;
+    /* FIXME sdolan: what's the right condition here? */
```

Comment:
An excellent question. I don't really believe in any of this caml_custom_get_max_major() stuff. "2/3 of 44% of the heap size, divided by 5". I keep expecting to subtract the number I first thought of and tada: it's the playing card hidden in the orange.

Reply:
Yeah, I think this criterion is wrong, and I believe so does @damiendoligez who changed it to something more sensible in his branch. I'm using the `caml_custom_get_max_major() / 5` number here to match the `Caml_state->extra_heap_resources > 0.2` logic it replaces, in order to make this PR preserve GC policy - I'd like to separate the rearrange-code and the change-policy parts of this work.

(The orange will be peeled in the next PR)